test/test_data/test_feature_validator.py

```diff
@@ -0,0 +1,355 @@
+        'list_numericalonly_nan',
+        'sparse',
+        'openml_40981',  # Australian
+    ),
+    indirect=True
+)
+def test_featurevalidator_supported_types(input_data_featuretest):
+    validator = FeatureValidator()
+    validator.fit(input_data_featuretest)
+    transformed_X = validator.transform(input_data_featuretest)
+    if sparse.issparse(input_data_featuretest):
+        assert sparse.issparse(transformed_X)
+    else:
+        assert isinstance(transformed_X, np.ndarray)
+    assert np.shape(input_data_featuretest) == np.shape(transformed_X)
+    assert np.issubdtype(transformed_X.dtype, np.number)
+    assert validator._is_fitted
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'list_categoricalonly_nonan',
+        'list_categoricalonly_nan',
+        'list_mixed_nonan',
+        'list_mixed_nan',
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_list(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r".*When providing a list of features.*Numpy failed"):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_string_nonan',
+        'numpy_string_nan',
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_numpy(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r".*When providing a numpy array.*not supported."):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'pandas_categoricalonly_nan',
+        'pandas_mixed_nan',
+        'openml_179',  # adult workclass has NaN in columns
+    ),
+    indirect=True
+)
+def test_featurevalidator_unsupported_pandas(input_data_featuretest):
+    validator = FeatureValidator()
+    with pytest.raises(ValueError, match=r"Categorical features in a dataframe.*missing/NaN"):
+        validator.fit(input_data_featuretest)
+
+
+@pytest.mark.parametrize(
+    'input_data_featuretest',
+    (
+        'numpy_categoricalonly_nonan',
+        'numpy_mixed_nonan',
+        'numpy_categoricalonly_nan',
+        'numpy_mixed_nan',
+        'pandas_categoricalonly_nonan',
+        'pandas_mixed_nonan',
+    ),
+    indirect=True
+)
+def test_featurevalidator_fitontypeA_transformtypeB(input_data_featuretest):
+    """
+    Check if we can fit in a given type (numpy) yet transform
+    if the user changes the type (pandas then)
+
+    This is problematic only in the case we create an encoder
+    """
+    validator = FeatureValidator()
+    validator.fit(input_data_featuretest)
+    if isinstance(input_data_featuretest, pd.DataFrame):
+        complementary_type = input_data_featuretest.to_numpy()
+    elif isinstance(input_data_featuretest, np.ndarray):
+        complementary_type = pd.DataFrame(input_data_featuretest)
+    validator.transform(complementary_type)
```

Comment:
Would it make sense to add a type check and a dimensions check here as above?

Reply:
Done!